pkg/alertmanager/multitenant.go

```diff
@@ -737,6 +744,18 @@ func (am *MultitenantAlertmanager) computeConfig(cfgs alertspb.AlertConfigDescs)
 
 // syncStates promotes/unpromotes the Grafana state and updates the 'promoted' flag if needed.
 func (am *MultitenantAlertmanager) syncStates(ctx context.Context, cfg amConfig) error {
+	// fetching grafana state first so we can register its size independently of it being promoted or not
+	s, err := am.store.GetFullGrafanaState(ctx, cfg.User)
+	if err != nil {
+		if errors.Is(err, alertspb.ErrNotFound) {
+			// This is expected if the state was already promoted.
+			level.Debug(am.logger).Log("msg", "grafana state not found, skipping promotion", "user", cfg.User)
```

Comment:
The metric is reporting the size of the last state even after we delete it. until the AM instance is restarted. We could add this line when we get an `ErrNotFound` to fix the issue:

```go
am.multitenantMetrics.grafanaStateSize.DeleteLabelValues(cfg.User)
```

This way, we delete the metric for the user once the state is gone from the DB.

Reply:
Ahh, good catch! Fixed now 👍